### DISCUSSION TITLE
How to define a field as a subset of an enum?

### DISCUSSION THREAD

**Discussion by berzi** [pydantic/pydantic#4232] [Question]
Let's say I have an enum:
```python
from enum import Enum, auto

class MyEnum(Enum):
    A = auto()
    B = auto()
    C = auto()
```

Now I want to make a model which will use it, but it should only accept a subset of the enum's members. If I type it like so:
```python
from typing import Union
from pydantic import BaseModel

class MyModel(BaseModel):
    my_field: Union[MyEnum.A, MyEnum.B]
```

When instantiating the model I get:
> pydantic.errors.ConfigError: field "type_ForwardRef(<MyEnum.A: 0>)" not yet prepared so type is still a ForwardRef, you might need to call MyModel.update_forward_refs().

So it seems using `Union` is not the way to go, although it's certainly what would be most obvious to me. Is there a way to do what I intend to here or should I submit a feature request?

**Comment by ghost** [comment]
You can use 
```
from typing import Literal
from pydantic import BaseModel

class MyModel(BaseModel):
    my_field: Literal[MyEnum.A, MyEnum.B]
```
